Sources/LanguageServerProtocol/CodeAction.swift

```diff
@@ -47,6 +48,30 @@ public struct CodeActionRequest: TextDocumentRequest, Hashable {
     self.context = context
     self.textDocument = textDocument
   }
+
+  public func injectMetadata(atResponse response: CodeActionRequestResponse?) -> CodeActionRequestResponse? {
+    let metadata = SourceKitLSPCommandMetadata(textDocument: textDocument)
+    guard let data = try? JSONEncoder().encode(metadata),
+          let metadataArgument = try? JSONDecoder().decode(CommandArgumentType.self, from: data) else
```

Comment:
We should just construct the dictionary instead of round-tripping through json.  The json (de) serialization code has a relatively high up front cost even for small payloads.

Reply:
Ah, I was trying to make the compiler synthetize the usage of CommandArguments as much as possible to make it easy to add and edit commands... I changed these guys to encode and decode the dictionaries manually, but let me know if there's something that can be done to reduce the boilerplate.

Reply:
I'm not aware of anything good here.  In principle you could write a Decoder implementation that works directly from the dictionary - basically perform the object mapping part of JSONDecoder without doing any of the parsing.  But that seems unnecessarily general to implement in sourcekit-lsp, so unless someone adds such a thing to the stdlib or Foundation I don't see us using it.